### PR TITLE
6021 ctrl click clip pitch or speed button resets value

### LINKED ISSUE #6021: Alt+click on pitch and speed clip buttons resets to default value
Interactive buttons were just introduced that provide a quick access to modifying pitch and speed of clips.
With tempo estimation, there will be situations where tempo is wrongly detected. We should then make it as easy as possible for users to reset to the default value.
The idea is that Alt-clicking these buttons does this. (This will also make the buttons disappear, as they only are visible when the value is non-default).

### PR DESCRIPTION
**Pull request by saintmatthieu**
Resolves: #6021 

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [ ] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA:
- [x] Ctrl+click on clip pitch or speed button resets the value to the default (causing the buttons to disappear)
- [x] Hovering over button shows tip in bottom left of Audacity window that Ctrl (Win, Linux) / Cmd (Mac) + click resets to default value.

Changing a clip's speed via the pitch-and-speed dialog or by ctrl-clicking the speed button ...
- [x] updates the selection to the new size of the clip (new behaviour)
- [x] throws a not-enough-room popup rather than overlapping with the following clip (like current master)
- [x] inserts an undo item in the project history, makes the project dirty and can be undone

### CODE REVIEW

**Comment by saintmatthieu** [comment]
@vsverchinsky I remember your remark of a recent PR of mine you reviewed. It was one large commit and was tedious to review.
I haven't gone as far as making sure that each commit builds this time, but at least I've created individual commits where I simply move code around.

File: src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp [hunk 1]
```diff
@@ -159,10 +159,19 @@ UIHandle::Result ClipPitchAndSpeedButtonHandle::DoRelease(
 HitTestPreview ClipPitchAndSpeedButtonHandle::Preview(
    const TrackPanelMouseState& state, AudacityProject* pProject)
 {
+   const auto ctrlDown = state.state.CmdDown();
+   const auto macOs =
+      wxPlatformInfo::Get().GetOperatingSystemFamilyName() == "Macintosh";
    if (mType == Type::Pitch)
-      return { XO("Click to change clip pitch."), nullptr };
+      return { ctrlDown ? XO("Click to reset clip pitch.") :
```
Comment (saintmatthieu) [inline src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp:166]: @LWinterberg your input please.

File: src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp [hunk 2]
```diff
@@ -159,10 +159,19 @@ UIHandle::Result ClipPitchAndSpeedButtonHandle::DoRelease(
 HitTestPreview ClipPitchAndSpeedButtonHandle::Preview(
    const TrackPanelMouseState& state, AudacityProject* pProject)
 {
+   const auto ctrlDown = state.state.CmdDown();
+   const auto macOs =
+      wxPlatformInfo::Get().GetOperatingSystemFamilyName() == "Macintosh";
```
Comment (LWinterberg) [inline src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp:164]: stray thought: Usually I see platform dependent stuff done as `#if __WXMAC__`

File: src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp (hunk 1 above)
Comment (LWinterberg) [reply src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp:166]: These look good to me

**Review by LWinterberg — APPROVED** [review]

File: src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp (hunk 2 above)
Comment (crsib) [reply src/tracks/playabletrack/wavetrack/ui/ClipPitchAndSpeedButtonHandle.cpp:164]: Or at least using [GetOperatingSystemId](https://docs.wxwidgets.org/3.0/classwx_platform_info.html#ab70a9c0bac9a38f05930e5f05a153cff)

**Review by vsverchinsky — APPROVED** [review]

**Pull request merged**